httpx/_models.py

```diff
@@ -1080,23 +1082,26 @@ def get(  # type: ignore
         default: str | None = None,
         domain: str | None = None,
         path: str | None = None,
+        raise_when_not_found: bool = False,
```

Comment:
Could we avoid adding this API as part of the pull request?

Reply:
Here, I wanted to preserve this responsibility of raising the `KeyError` exception to the `__getitem__()` method.

However, if it's safe to raise it whenever it's not found through `get()`, then we could remove `raise_when_not_found` parameter.

What do you think?